src/vehicle/vehicle.go

```diff
@@ -0,0 +1,74 @@
+package vehicle
+
+import (
+	"fmt"
+
+	"github.com/HyperloopUPV-H8/Backend-H8/common"
+	"github.com/HyperloopUPV-H8/Backend-H8/packet_parser"
+	"github.com/HyperloopUPV-H8/Backend-H8/pipe"
+	"github.com/HyperloopUPV-H8/Backend-H8/sniffer"
+	"github.com/HyperloopUPV-H8/Backend-H8/unit_converter"
+	"github.com/HyperloopUPV-H8/Backend-H8/vehicle/internals"
+	"github.com/HyperloopUPV-H8/Backend-H8/vehicle/models"
+)
+
+type Vehicle struct {
+	sniffer          *sniffer.Sniffer
+	parser           *packet_parser.PacketParser
```

Comment:
Why are they pointers?

Reply:
because AddGlobal and AddPacket work on a pointer to the type